FortnoxAPILibrary/UrlRequestBase.cs

```diff
@@ -92,14 +104,14 @@ internal string GetUrl(string index = "")
         internal HttpWebRequest SetupRequest(string requestUriString, string method)
         {
             Error = null;
-            if (string.IsNullOrEmpty(ConnectionCredentials.AccessToken) || string.IsNullOrEmpty(ConnectionCredentials.ClientSecret))
+            if (string.IsNullOrEmpty(_accessToken) || string.IsNullOrEmpty(_clientSecret))
             {
                 throw new Exception("Access-Token and Client-Secret must be set");
             }
 
             HttpWebRequest wr = (HttpWebRequest)HttpWebRequest.Create(requestUriString);
-            wr.Headers.Add("access-token", ConnectionCredentials.AccessToken);
-            wr.Headers.Add("client-secret", ConnectionCredentials.ClientSecret);
+            wr.Headers.Add("access-token", _accessToken);
```

Comment:
You could set the ConnectionCredentials to be the back up value if the credentials wasn't specified during the creation of the connector. To maintain backwards compatibility an empty constructor would be required for the UrlRequestBase class and the classes that inherit from the UrlRequestBase class.

How about something like this?

``` C#
wr.Headers.Add("access-token", _accessToken ?? ConnectionCredentials.AccessToken);
```